packages/synapse-react-client/src/components/DownloadCart/DownloadListTable.tsx

```diff
@@ -112,7 +300,62 @@ export default function DownloadListTable(props: DownloadListTableProps) {
     }
   }, [isError, newError, handleError])
 
-  const allRows = data?.pages.flatMap(page => page.page) ?? []
+  const { mutateAsync: removeFileFromDownloadList } =
+    useRemoveFilesFromDownloadList()
+
+  const removeItem = useCallback(
+    async (
+      item: DownloadListItem,
+      fileName: string,
+      notificationTitle: string,
+    ) => {
+      await removeFileFromDownloadList({ batchToRemove: [item] })
+      notifyRemovedFromDownloadList(notificationTitle, fileName)
+    },
+    [removeFileFromDownloadList],
+  )
+
+  const allRows = useMemo(
+    () => data?.pages.flatMap(page => page.page) ?? [],
+    [data?.pages],
+  )
+
+  const columns = useMemo(
+    () =>
+      getColumns({
+        removeItem: (...args) => {
+          void removeItem(...args)
+        },
+        onCopySynIds: () => {
+          // trigger loading all pages of the download list table, and then copy all IDs to the clipboard
+          setCopyingAllSynapseIDs(true)
+        },
+      }),
+    [removeItem],
+  )
+
+  const table = useReactTable<DownloadListItemResult>({
+    data: allRows,
+    columns: columns,
+    getCoreRowModel: getCoreRowModel(),
+    manualSorting: true,
+    onSortingChange: setTableSortState,
+    state: {
+      columnPinning: {
+        right: ['actions'],
+      },
```

Comment:
TanStack table has built-in support for `columnPinning`, which is basically the same thing as our static "Actions" column. The "pinned" columns will not change for this table, so this state value is constant.

Reply:
nice!